product/src/it/java/com/yas/product/controller/ProductControllerIT.java

```diff
@@ -0,0 +1,526 @@
+import com.yas.product.viewmodel.product.ProductPutVm;
+import com.yas.product.viewmodel.product.ProductQuantityPostVm;
+import com.yas.product.viewmodel.product.ProductQuantityPutVm;
+import com.yas.product.viewmodel.product.ProductVariationPostVm;
+import com.yas.product.viewmodel.product.ProductVariationPutVm;
+import com.yas.product.viewmodel.productoption.ProductOptionValuePostVm;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
+
+@Slf4j
+@Import(IntegrationTestConfiguration.class)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ProductControllerIT extends AbstractControllerIT {
+
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private ProductService productService;
+    @Autowired
+    private BrandRepository brandRepository;
+    @Autowired
+    private CategoryRepository categoryRepository;
+    @Autowired
+    private ProductCategoryRepository productCategoryRepository;
+    @Autowired
+    private ProductImageRepository productImageRepository;
+    @Autowired
+    private ProductOptionRepository productOptionRepository;
+    @Autowired
+    private ProductRelatedRepository productRelatedRepository;
+
+    private Product productOne;
+    private Product productTwo;
+    private Product product;
+    private ProductRelated productRelated;
+    private ProductOption productOption;
+    private ProductQuantityPostVm productQuantityPostVm;
+    private ProductQuantityPutVm productQuantityPutVm;
+    private Brand brand;
+    private Category category;
+    private ProductCategory productCategory;
+    private ProductVariationPostVm productVariationPostVm;
+    private ProductVariationPutVm productVariationPutVm;
+    private ProductOptionValuePostVm productOptionValuePostVm;
+    private ProductPostVm productPostVm;
+    private ProductPutVm productPutVm;
+
+    private static final String PRODUCT_BACKOFFICE_URL = "/v1/backoffice/products";
+    private static final String PRODUCT_STOREFRONT_URL = "/v1/storefront/products";
+
+    @BeforeEach
+    public void insertTestData() {
+        brand = new Brand();
+        brand.setName("Brand");
+        brand.setSlug("brand-slug");
+        brandRepository.save(brand);
+
+        category = new Category();
+        category.setName("Category");
+        categoryRepository.save(category);
+
+        productOne = Product.builder()
```

Comment:
Extract test data object creations into another helper class to increase the readability of `insertTestData()`

Reply:
I’ve separated it. Please review it. Thank you!